graspologic/utils/utils.py

```diff
@@ -514,11 +523,39 @@ def largest_connected_component(graph, return_inds=False):
     lcc.remove_nodes_from([n for n in lcc if n not in lcc_nodes])
     if return_inds:
         nodelist = np.array(list(lcc_nodes))
-    if input_ndarray:
-        lcc = nx.to_numpy_array(lcc)
     if return_inds:
         return lcc, nodelist
-    return lcc
+    else:
+        return lcc
+
+
+def _largest_connected_component_adjacency(
+    adjacency: Union[np.ndarray, csr_matrix],
+    return_inds: bool = False,
+):
+    # If you treat an undirected graph as directed and take the largest weakly connected
+    # component, you'll get the same answer as taking the largest connected component of
+    # that undirected graph. So I wrote it this way to avoid the cost of checking for
+    # directedness, and it makes the code simpler too.
+    n_components, labels = csgraph.connected_components(
+        adjacency, directed=True, connection="weak", return_labels=True
```

Comment:
Just for myself; do you never look for strongly connected components?  Weakly seems to make sense and I don't see why it wouldn't be the default, but I just wanted to ask and make sure I wasn't silly

Reply:
I think for most spectral stuff weakly connected just happens to be the definition we care about

There are other applications (like markov chain stuff) where you'd definitely want something to be strongly connected. I just havent seen it come up in this package yet. I suppose we are just being opinionated about what we mean for now? If we wanted to make connection='strong'/'weak' an option then we could talk about it